### PR TITLE
Print status output directly to stdout without 'logger' prefixes

### PR DESCRIPTION
**Pull request by scottopell**
### What does this PR do?
This changes the behavior of the `agent status` command to no longer use the `logger` component to do the printing.

Before:
```
./bin/agent/agent -c ~/dev/datadog-config-scenarios/tcp-log-listener/ status -j --pretty-json | head
2022-08-15 13:36:56 EDT | CORE | WARN | (pkg/util/log/log.go:616 in func1) | Deactivating Autoconfig will disable most components. It's recommended to use autoconfig_exclude_features and autoconfig_include_features to activate/deactivate features selectively
2022-08-15 13:36:56 EDT | CORE | INFO | (cmd/system-probe/config/config.go:119 in Merge) | no config exists at /etc/datadog-agent/system-probe.yaml, ignoring...
2022-08-15 13:36:56 EDT | CORE | INFO | (cmd/agent/app/status.go:165 in requestStatus) | {
  "JMXStartupError": {
    "LastError": "",
    "Timestamp": 0
  },
  "JMXStatus": {
    "checks": {
      "failed_checks": null,
```

After:
```
./bin/agent/agent -c ~/dev/datadog-config-scenarios/tcp-log-listener/ status -j | jq '. |{hostnameStats}'
{
  "hostnameStats": {
    "errors": {
      "'hostname' configuration/environment": "hostname is empty",
      "'hostname_file' configuration/environment": "'hostname_file' configuration is not enabled",
      "aws": "not retrieving hostname from AWS:START - datadog.apiKey=***************************13add - END the host is not an ECS instance and other providers already retrieve non-default hostnames",
      "azure": "azure_hostname_style is set to 'os'",
      "container": "the agent is not containerized",
      "fargate": "Agent is not running on Fargate START - datadog.apiKey=***************************13add - END",
      "fqdn": "'hostname_fqdn' configuration is not enabled",
      "gce": "unable to retrieve hostname from GCE: GCE metadata API error: Get \"http://169.254.169.254/computeMetadata/v1/instance/hostname\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"
    },
    "provider": "os"
  }
}
```

### Motivation
We want to be able to specify `-j` to get json output and pipe directly into something that can process json.

In 7.38 and 7.39-rc, this would fail as the `status` output is printed by the `logger` so it has timestamp and other front-matter.

### Additional Notes
The `status` output was previously printed in this same way, directly to stdout using `fmt.Print`, however in https://github.com/DataDog/datadog-agent/pull/12338 I changed this to use the `logger` component in order to take advantage of the "automatic scrubbing" that the logger has.

This had the side effect of changing the default output, and led to other followup issues such as https://github.com/DataDog/datadog-agent/pull/12338 where now log levels are taken into account.

This approach is simpler and closer to the historical behavior of `agent status` ([before any 'scrubber' changes](https://github.com/DataDog/datadog-agent/blob/c0768f4d17511c372be1d4167689a5a3f50ad631/cmd/agent/app/status.go))

### Describe how to test/QA your changes
N/A

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by djmitche — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]
LGTM. Can we add a changelog for it though? I don't think it's strictly necessary but it would be a bugfix that users may want to know about if they depended on this feature.

File: releasenotes/notes/plain-json-agent-status-5d0a427ae6cc2338.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    'agent status' command output is now parseable as json
```
Comment (maycmlee) [inline releasenotes/notes/plain-json-agent-status-5d0a427ae6cc2338.yaml:11]:
```suggestion
    'agent status' command output is now parseable as JSON
```

File: releasenotes/notes/plain-json-agent-status-5d0a427ae6cc2338.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    directly from stdout. Before this change there was some
```
Comment (maycmlee) [inline releasenotes/notes/plain-json-agent-status-5d0a427ae6cc2338.yaml:12]:
```suggestion
    directly from stdout. Before this change, the
```

File: releasenotes/notes/plain-json-agent-status-5d0a427ae6cc2338.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    logger front-matter making it hard to parse 'status'
```
Comment (maycmlee) [inline releasenotes/notes/plain-json-agent-status-5d0a427ae6cc2338.yaml:13]:
```suggestion
    logger front-matter made it hard to parse 'status'
```

**Review by maycmlee — COMMENTED** [review]
Left some feedback

**Review by maycmlee — APPROVED** [review]
👍 for docs.
